src/nodes.coffee

```diff
@@ -2904,12 +2903,14 @@ exports.Param = class Param extends Base
   # `name` is the name of the parameter and `node` is the AST node corresponding
   # to that name.
   eachName: (iterator, name = @name) ->
-    atParam = (obj) => iterator "@#{obj.properties[0].name.value}", obj, @
+    atParam = (obj) => iterator "@#{obj.properties[0].name.value}", obj, @, nObj
     # * simple literals `foo`
     return iterator name.value, name, @ if name instanceof Literal
     # * at-params `@foo`
     return atParam name if name instanceof Value
     for obj in name.objects ? []
+      # Save original obj.
+      nObj = obj
```

Comment:
why?

Reply:
In case of object destructuring parameter with the default value, e.g. `({@works = "no", @drinks}) ->`,  when replacement function is called from `@renameParam`, the `parent` is `Assign` in case of `@works = "no"`, and `Obj` in case of `@drinks`.
Because of this `@works = "no"` is changed to `works1 = "no"` and `@drinks` to `drinks:drinks1`.

The best way, I could find, to solve this, was to pass  `Assign` (`{works:works1}`)  to  `@renameParam` instead of `Identifier` (`works1`).
But, because `Param::eachName` passes `obj.variable`  (`@works` in this case) to the iterator,   I had to find a way to detect if param has a default value set in order to create a correct replacement. Passing original `obj` seems like a way to go.
Another way would be passing index and use it to check the argument in the `Code::eachParamName` (e.g.  `param.name.parameters[i]`).

Reply:
Ok, now I get it. I don't like that `atParam` magically references a variable from the for loop, tbh. Can you please make it explicit?

Reply:
You mean to pass it as an argument?
```coffeescript
atParam = (obj, originalObject = no) => iterator "@#{obj.properties[0].name.value}", obj, @, originalObject

....

else if obj.this
  atParam obj, nObj

```

Reply:
Yes.